File: esercizi_java/test/compito_python_2/Cliente.py
from Servizio import Servizio
class Cliente:
    def __init__(self, cod, nome):
        self.cod = cod
        self.nome = nome
        self.servizi = []


    def addServizio(self, d, ore):
        self.servizi.append(Servizio(d,ore))


    def numServ(self, codice):
        a = 0

        for s in self.servizi:
            if codice == s.dipendente.cod:
                a = a + 1
        return a


    def getCost(self):
        c=0
        for s in self.servizi: 
            c = s.ore*s.dipendente.costo+c
        return c

    def toString(self):
        return str(self.cod) + "\t"+self.nome + "\t" + str(self.getCost())

File: esercizi_java/NetBeansProjects/Palestra/src/palestra/Cliente.java
 package palestra;
import java.util.LinkedList;
import java.util.List;

public class Cliente {
    private int cod;
    private String nome;
    private List<Servizi> servizio;
    
    public Cliente(int cod, String nome){
        this.cod=cod;
        this.nome=nome;
        this.servizio = new LinkedList<Servizi>(); 
    }
    
    public void addServizio(Dipendente d, double ore){
        servizio.add(new Servizi(d,ore));
    }
    
    public double getcost(){
        double c=0;
        
        for(Servizi s: servizio){
            c=s.ore*s.d.cost+c;
        }
        
        return c;
    }
    
    public int numServ(int codice){
        int a=0;
        
        for(Servizi s: servizio){
            if(codice==s.d.cod){
                a=a+1;
            }
        }
        return a;
    }
    
    public String toString(){
        return cod + "\t"+ nome+ "\t"+getcost();
    }
}
